tasks/delete_old_reconcile_jobs/delete_old_reconcile_jobs.py

```diff
@@ -0,0 +1,248 @@
+        db_connect_info: See shared_db.py's get_configuration for further details.
+    """
+
+    user_engine = shared_db.get_user_connection(db_connect_info)
+    old_job_ids = get_jobs_older_than_x_days(
+        internal_reconciliation_expiration_days, user_engine
+    )
+
+    if old_job_ids is None:
+        LOGGER.info("No jobs found.")
+        return
+
+    delete_jobs(
+        old_job_ids,
+        user_engine,
+    )
+
+
+@shared_db.retry_operational_error()
+def get_jobs_older_than_x_days(
+    internal_reconciliation_expiration_days: int, engine: Engine
+) -> List[int]:
+    """
+    Gets all jobs older than internal_reconciliation_expiration_days days.
+    If none are found, returns null.
+
+    Args:
+        internal_reconciliation_expiration_days: Only reports updated before this many days ago will be retrieved.
+        engine: The sqlalchemy engine to use for contacting the database.
+
+    Returns: A list of ids for the jobs, or null if none were found.
+
+    """
+    try:
+        LOGGER.debug(
+            f"Getting jobs older than {internal_reconciliation_expiration_days} days."
+        )
+        with engine.begin() as connection:
+            sql_results = connection.execute(
+                # populate reconcile_phantom_report with files in orca.files but NOT reconcile_s3_object
+                get_jobs_sql(),
+                [
+                    {
+                        "internal_reconciliation_expiration_days": internal_reconciliation_expiration_days
+                    }
+                ],
+            )
+            return sql_results.fetchone()[
+                0
+            ]  # fetchone returns row. [0] returns the list of ids.
+    except Exception as sql_ex:
+        LOGGER.error(f"Error while getting jobs: {sql_ex}")
+        raise
+
+
+@shared_db.retry_operational_error()
+def delete_jobs(job_ids: List[int], engine: Engine) -> None:
+    """
+    Deletes all records for the given job ids from the database.
+
+    Args:
+        job_ids: The ids of the jobs containing s3 inventory info.
+        engine: The sqlalchemy engine to use for contacting the database.
+    """
+    try:
+        LOGGER.debug(f"Deleting data for job ids {job_ids}.")
+        with engine.begin() as connection:
+            connection.execute(
+                # populate reconcile_phantom_report with files in orca.files but NOT reconcile_s3_object
+                delete_jobs_sql(),
+                [{"job_ids_to_delete": job_ids}],
+            )
+    except Exception as sql_ex:
+        LOGGER.error(f"Error while deleting jobs: {sql_ex}")
+        raise
+
+
+def get_jobs_sql() -> TextClause:
+    """
+    SQL for getting jobs older than a certain date.
+    """
+    return text(
+        """
+        SELECT json_agg(reconcile_job.id)
+            FROM reconcile_job
+            WHERE last_update < (NOW() - interval ':internal_reconciliation_expiration_days days')"""
+        # todo: limit?
```

Comment:
Do we want to limit this? It'd be a pretty extreme case where the number of jobs would be prohibitive, but it might be a good practice.